src/cdk/tree/tree.ts

```diff
@@ -0,0 +1,272 @@
+  /** Subject that emits when the component has been destroyed. */
+  private _destroyed = new Subject<void>();
+
+  /** Latest data provided by the data source through the connect interface. */
+  private _data: NgIterable<T> = [];
+
+  /** Differ used to find the changes in the data provided by the data source. */
+  private _dataDiffer: IterableDiffer<T>;
+
+  /** Stores the node definition that does not have a when predicate. */
+  private _defaultNodeDef: CdkNodeDef<T> | null;
+
+  /** Focus related key manager */
+  _keyManager: FocusKeyManager<CdkTreeNode<T>>;
+
+  /** For focus, ordered nodes */
+  orderedNodes: QueryList<CdkTreeNode<T>> = new QueryList<CdkTreeNode<T>>();
+
+  /**
+   * Provides a stream containing the latest data array to render. Influenced by the tree's
+   * stream of view window (what nodes are currently on screen).
+   */
+  @Input()
+  get dataSource(): DataSource<T> { return this._dataSource; }
+  set dataSource(dataSource: DataSource<T>) {
+    if (this._dataSource !== dataSource) {
+      this._switchDataSource(dataSource);
+    }
+  }
+  private _dataSource: DataSource<T>;
+
+  /** The tree controller */
+  @Input() treeControl: TreeControl<T>;
+
+  // TODO(andrewseguin): Remove max value as the end index
+  //   and instead calculate the view on init and scroll.
+  /**
+   * Stream containing the latest information on what nodes are being displayed on screen.
+   * Can be used by the data source to as a heuristic of what data should be provided.
+   */
+  viewChange =
+    new BehaviorSubject<{start: number, end: number}>({start: 0, end: Number.MAX_VALUE});
+
+  // Outlets within the tree's template where the nodes will be inserted.
+  @ViewChild(NodeOutlet) _nodeOutlet: NodeOutlet;
+
+  /** The tree node template for the tree */
+  @ContentChildren(CdkNodeDef) _nodeDefs: QueryList<CdkNodeDef<T>>;
+
+  /** The tree node inside the tree */
+  @ContentChildren(CdkTreeNode, {descendants: true}) items: QueryList<CdkTreeNode<T>>;
+
+  constructor(private _differs: IterableDiffers,
+              private _changeDetectorRef: ChangeDetectorRef) {}
+
+  ngOnInit() {
+    this._dataDiffer = this._differs.find([]).create();
+    if (!this.treeControl) {
+      throw getTreeControlMissingError();
+    }
+  }
+
+  ngOnDestroy() {
+    this._nodeOutlet.viewContainer.clear();
+
+    this._destroyed.next();
+    this._destroyed.complete();
+
+    if (this.dataSource) {
+      this.dataSource.disconnect(this);
+    }
+  }
+
+  ngAfterContentChecked() {
+    const defaultNodeDefs = this._nodeDefs.filter(def => !def.when);
+    if (defaultNodeDefs.length > 1) { throw getTreeMultipleDefaultNodeDefsError(); }
+    this._defaultNodeDef = defaultNodeDefs[0];
+
+    if (this.dataSource) {
+      this._observeRenderChanges();
+    }
+  }
+
+  ngAfterViewInit() {
+    // For key traversal in correct order
+    this.items.changes.subscribe(items => {
```

Comment:
It seems like you're never unsubscribing from this one and I'm not sure whether Angular will complete the stream on destroy.